nova/virt/vmwareapi/cluster_util.py

```diff
@@ -78,27 +74,13 @@ def create_group_spec(client_factory, group, operation):
     return group_spec
 
 
-def _create_vm_group_spec(client_factory, group_info, vm_refs,
-                          operation="add", group=None):
-    if group:
-        # On vCenter UI, it is not possible to create VM group without
-        # VMs attached to it. But, using APIs, it is possible to create
-        # VM group without VMs attached. Therefore, check for existence
-        # of vm attribute in the group to avoid exceptions
-        if hasattr(group, 'vm'):
-            vm_refs = vm_refs + group.vm
-
-    group = create_vm_group(client_factory, group_info.name, vm_refs, group)
-
-    return create_group_spec(client_factory, group, operation)
-
-
-def _get_vm_group(cluster_config, group_info):
+def _get_vm_group(cluster_config, group_name):
     if not hasattr(cluster_config, 'group'):
-        return
+        return None
```

Comment:
Why those explicit `return None` instead of just `return`?

Reply:
My linter says:
> According to PEP8, if any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)

Presumably, the idea is to avoid accidentally returning `None` by just missing to type the returned value, or to write a return statement at the end, or when refactoring code missing to adopt all the code to return a value.

I think, I run into each of those cases, hence the linter, and me preferring to follow that rule.

Reply:
Never heard of that. It does make somewhat sense, but imho adds clutter, too. Anyways, let's keep your linter happy then.